### PR TITLE
Change of email process security updates #1140

### PR DESCRIPTION
**Pull request by dvlshah**
### Proposed changes in this pull request

Fixed(#1140) except "user's email address should not be updated until verification"

- The confirmation email sent to the new email address does not include the username as mentioned for security purposes.

/cadasta/templates/allauth/account/email/email_confirmation_message.txt 

User Display is removed which was displaying username in verification mail sent to new email id.

-  A separate notification email is sent to the old address. If the event of unauthorized changes occurs, this will alert the user to the change.The notification email to the old email addresses advise the user to let us know immediately if this change was not authorized, and contact method is provided in email itself.

/cadasta/accounts/forms.py --> clean_email() method 

- Added  new test method in /cadasta/accounts/tests/test_forms.py where I assign assert len(mail.outbox) == 2 in  "test_update_user" method  because we are sending mails to old user account as well as new user account.

### When should this PR be merged

I have run './runtests.py' in vagrant directory, and the all tests had ran successfully. So once it's been reviewed there will not be any conflict hence it can be merged.

I have manually tried different email updates and everytime it gives expected behavior as mentioned above.

### Checklist (for reviewing)
@oliverroick @wonderchook 
#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: cadasta/accounts/forms.py [hunk 1]
```diff
@@ -89,7 +89,14 @@ def clean_email(self):
         email = self.data.get('email')
         if self.instance.email != email:
             self._send_confirmation = True
-
+            send_mail(
+                "Email Update",
+                "Email Changed to "+email+" for your cadasta account." +
+                "Let us know if it was not you at"+settings.DEFAULT_FROM_EMAIL,
+                settings.DEFAULT_FROM_EMAIL,
+                [email],
```
Comment (oliverroick) [inline cadasta/accounts/forms.py]: The email is sent to the new email address and not the old one.

File: cadasta/accounts/forms.py [hunk 2]
```diff
@@ -89,7 +89,14 @@ def clean_email(self):
         email = self.data.get('email')
         if self.instance.email != email:
             self._send_confirmation = True
-
+            send_mail(
```
Comment (oliverroick) [inline cadasta/accounts/forms.py]: You place `send_mail` into `clean_email`. `clean_email` is used to check if the new email address is already used by a another user. You even send the email _before_ this check, so every time someone attempts to change the email address, the info email is sent, whether the update of the address is successful or not. 

The email should be sent only if the email has been updated, i.e. only when the new email has been accepted.

File: cadasta/templates/allauth/account/email/email_confirmation_message.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
-{% load account %}{% user_display user as user_display %}{% load i18n %}{% autoescape off %}{% blocktrans %}
+{% load account %}{% load i18n %}{% autoescape off %}{% blocktrans %}
 
-You're receiving this email because user {{ user_display }} at Cadasta Platform has given yours as an email address to connect their account.
+You're receiving this email because user at Cadasta Platform has given yours email-id as an email address to connect their account.
```
Comment (oliverroick) [inline cadasta/templates/allauth/account/email/email_confirmation_message.txt]: I would reword this as "You're receiving this email because a user at Cadasta Platform has this email address email address to connect their account."

Comment (dvlshah) [inline cadasta/templates/allauth/account/email/email_confirmation_message.txt]: "You're receiving this email because a user at Cadasta Platform has provided this email address as a email address to connect their account."
Is this correct way to write it ?

File: cadasta/accounts/forms.py (hunk 2 above)
Comment (dvlshah) [inline cadasta/accounts/forms.py]: Can you guide me from where to send mail to old user account alerting changes are being made?

File: cadasta/templates/allauth/account/email/email_confirmation_message.txt (hunk 3 above)
Comment (oliverroick) [inline cadasta/templates/allauth/account/email/email_confirmation_message.txt]: It's what I suggest; I suppose it's correct. Is anything unclear that you would change?

File: cadasta/accounts/forms.py (hunk 2 above)
Comment (oliverroick) [inline cadasta/accounts/forms.py]: The email needs to be sent _after_ the validation is run successfully. Have a look at the form you're working in. Another email is already sent, sending the email to the old address should work similar. I don't want to provide too much guidance because we want to see whether you can solve problems yourself.

File: cadasta/templates/allauth/account/email/email_confirmation_message.txt (hunk 3 above)
Comment (dvlshah) [inline cadasta/templates/allauth/account/email/email_confirmation_message.txt]: No, Actually I misread . Sorry for inconvenience. I have updated all changes and I will commit asap.Thanks for valuable inputs.
